### PR TITLE
Add story for branded Panel

### PR DESCRIPTION
**Pull request by umpox**
Main aim here is to get the Panel scaffolding covered by our visual regression tests, especially given future redesign work may touch these parts.

Part of https://github.com/sourcegraph/sourcegraph/issues/19520

image

Going to look at catching more typical panels in Percy as part of a different PR!

### CODE REVIEW

File: client/branded/src/components/panel/Panel.story.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                                menus: {
+                                    'panel/toolbar': [
+                                        {
+                                            action: 'a',
+                                        },
+                                        {
+                                            action: 'b',
+                                        },
+                                    ],
+                                },
```
Comment (valerybugakov) [inline client/branded/src/components/panel/Panel.story.tsx:106]: nit: what do you think about extracting `menus` and `actions` into a standalone variables similar to `panels` for readability and consistency?

File: client/branded/src/components/panel/Panel.story.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    getPanelViews: () => pretendProxySubscribable(of(panels)),
+                    getActiveCodeEditorPosition: () => pretendProxySubscribable(of(null)),
+                })
+            ),
+        }}
+    />
+)
+
+add('Simple', () => (
+    <MemoryRouter initialEntries={[{ pathname: '/', hash: `#tab=${panels[0].id}` }]}>
```
Comment (valerybugakov) [inline client/branded/src/components/panel/Panel.story.tsx:120]: Is there any reason why do we want to duplicate the `MemoryRouter` for each variant? I've tried to move it to the `addDecorator` wrapper, and it seems to work.

File: client/branded/src/components/panel/Panel.story.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+add('With actions', () => (
+    <MemoryRouter initialEntries={[{ pathname: '/', hash: `#tab=${panels[0].id}` }]}>
+        <div>{PanelWithActions}</div>
+    </MemoryRouter>
+))
+
+add('With actions mobile', () => (
+    <MemoryRouter initialEntries={[{ pathname: '/', hash: `#tab=${panels[0].id}` }]}>
+        <div style={{ width: 320 }}>{PanelWithActions}</div>
```
Comment (valerybugakov) [inline client/branded/src/components/panel/Panel.story.tsx:143]: I saw in other stories the pattern to use [viewports](https://www.chromatic.com/docs/viewports) parameters to tell Chromatic about a responsive component.

https://github.com/sourcegraph/sourcegraph/blob/ad77440484c7d6064b1b85fa69d1e85857dd0dba/client/web/src/enterprise/batches/preview/list/PreviewList.story.tsx#L10-L16

File: client/branded/src/components/panel/Panel.tsx [hunk 4]
```diff
@@ -196,31 +193,32 @@ export const Panel = React.memo<Props>(props => {
...
     useEffect(() => {
         const subscription = registerPanelToolbarContributions(props.extensionsController.extHostAPI)
-
         return () => subscription.unsubscribe()
     }, [props.extensionsController])
 
     const handleActiveTab = useCallback(
         (index: number): void => {
-            history.replace(`${pathname}${currentTabLabel}=${items[index].id}`)
+            history.replace(`${location.pathname}${currentTabLabel}=${items[index].id}`)
```
Comment (valerybugakov) [inline client/branded/src/components/panel/Panel.tsx:214]: nit: is this change required because we need to pass the whole `location` object to the `PanelView` component? Would be cool to preserve shorter version for complex interpolations like this.

```ts
const location = useLocation()
const { hash, pathname } = location

history.replace(`${pathname}${currentTabLabel}=${items[index].id}`)

**Review by valerybugakov — APPROVED** [review]
One step closer to safety 🙌

File: client/branded/src/components/panel/Panel.story.tsx (hunk 2 above)
Comment (umpox) [reply client/branded/src/components/panel/Panel.story.tsx:120]: No particular reason, will update :)

File: client/branded/src/components/panel/Panel.tsx (hunk 4 above)
Comment (umpox) [reply client/branded/src/components/panel/Panel.tsx:214]: Yeah, I'll preserve this
